PyRoute/TradeCalculation.py

```diff
@@ -679,10 +680,10 @@ def route_update_simple(self, route):
             end.tradeOver += tradeCr if end != route[-1] else 0
             end.tradeCount += 1 if end != route[-1] else 0
             end.passOver += tradePass if end != route[-1] else 0
-            self.galaxy.stars[start][end]['trade'] += tradeCr
-            self.galaxy.stars[start][end]['count'] += 1
-            self.galaxy.stars[start][end]['weight'] -= \
-                self.galaxy.stars[start][end]['weight'] / self.route_reuse
+            data = self.galaxy.stars[start][end]
+            data['trade'] += tradeCr
+            data['count'] += 1
+            data['weight'] -= (data['weight'] - data['distance']) / self.route_reuse
```

Comment:
You have changed the formula for calculating route weights. Why? 

I'm rethinking the route reuse in this whole process. The system is inconsistent about which form of reuse is used, and it doesn't have has much effect (or has too much of an effect) on updating the routes. Especially with the changes to the route finding process.

Reply:
One of A*'s underlying requirements is that whatever heuristic is used _never_ over-estimate the cost to the target node.  

After paring off all the fancy bits, **Star.distanceHeuristic** uses the straight-line hex-distance between the end nodes.

Consider a link between A and B, 3 parsecs apart (ie, distance = 3).  The heuristic estimate for that link would be 3.  What happens under the old code when the **actual** link weight is already 3 and it comes time to update?  Right, the heuristic estimate between A and B becomes inadmissible as it over-estimates the cost.

As far as I can tell, this admissibility fix is _orthogonal_ to route reuse - which just determines how quickly the cost deflates.

Admissibility is a **much** bigger deal in the bidirectional version of pathfinding that I'm planning to land later on.